thunder/executors/cudnnex.py

```diff
@@ -42,783 +37,51 @@ def required_cudnn_version() -> LooseVersion:
-# and dtypes. They don't have to be passed in as `torch.Tensor`s.
-def _allocate_catted_grad_qkv(
-    query: torch.Tensor,
-    key: torch.Tensor,
-    value: torch.Tensor,
-) -> torch.Tensor:
-    assert _same_size_except(query.size(), key.size(), value.size(), except_dim=1)
-    assert query.dtype == key.dtype == value.dtype
-    assert query.device == key.device == value.device
-
-    b, s, d = query.size(0), query.size(2), query.size(3)
-    h_q, h_k, h_v = query.size(1), key.size(1), value.size(1)
-    h_qkv = h_q + h_k + h_v
-
-    # Create grad_qkv as a tensor of size [b,h_qkv,s,d] and allocation order
-    # [0,2,1,3] from major to minor.
-    return torch.empty(b, s, h_qkv, d, dtype=query.dtype, device=query.device).permute(0, 2, 1, 3)
-
-
-def _cudnn_sdpa_bwd_impl(
-    grad_out: torch.Tensor,
-    query: torch.Tensor,
-    key: torch.Tensor,
-    value: torch.Tensor,
-    attn_mask: None | torch.Tensor,
-    dropout_p: float,
-    is_causal: bool,
-    out: torch.Tensor,
-    softmax_stats: torch.Tensor,
-    philox_seed: torch.Tensor,
-    philox_offset: torch.Tensor,
-    *,
-    scale: None | float = None,
-    cat_grad_qkv: bool,
-) -> tuple[torch.Tensor, ...]:
-    query = _sdpa_enforce_input_tensor_contiguity(query)
-    key = _sdpa_enforce_input_tensor_contiguity(key)
-    value = _sdpa_enforce_input_tensor_contiguity(value)
-
-    # When cat_grad_qkv is on, allocate dQKV and make dQ, dK, and dV
-    # slices of that. Otherwise, allocate them individually.
-    grad_qkv: None | torch.Tensor = None
-    if cat_grad_qkv:
-        grad_qkv = _allocate_catted_grad_qkv(query, key, value)
-        grad_query, grad_key, grad_value = grad_qkv.split([query.size(1), key.size(1), value.size(1)], dim=1)
-    else:
-        grad_query = torch.empty_like(query)
-        grad_key = torch.empty_like(key)
-        grad_value = torch.empty_like(value)
-
-    if attn_mask is not None:
-        if query.ndim > attn_mask.ndim:
-            attn_mask = attn_mask.view(*((1,) * (query.ndim - attn_mask.ndim)), *attn_mask.shape)
-        if attn_mask.dtype == torch.bool:
-            attn_bias = torch.zeros_like(attn_mask, dtype=query.dtype)
-            attn_bias.masked_fill_(attn_mask.logical_not(), float("-inf"))
-            attn_mask = attn_bias
+    sdpa_impl.cudnn = cudnn
 
-    (
-        Q,
-        K,
-        V,
-        O,
-        dO,
-        Stats,
-        Seed,
-        Offset,
-        Attn_scale,
-        Bias,
-        dQ,
-        dK,
-        dV,
-        dBias,
-        graph,
-    ) = _make_cudnn_sdpa_backward_graph(
-        query,
-        key,
-        value,
-        attn_mask,
-        dropout_p,
-        is_causal,
-        query.stride(),
-        key.stride(),
-        value.stride(),
-        grad_query.stride(),
-        grad_key.stride(),
-        grad_value.stride(),
+    cudnn_ex = OperatorExecutor("cudnn", version=cudnn.backend_version())
+    register_executor(cudnn_ex)
```

Comment:
Almost there! We don't actually need to register the executor here, we can keep the registration in the sdpa file and then just export the symbols from there.

```suggestion
    import thunder.executors.cudnn_sdpa as sdpa_impl

    cudnn_ex = sdpa_impl.cudnn_ex
```